test/spec/ol/renderer/canvas/imagelayer.test.js

```diff
@@ -79,81 +86,74 @@ describe('ol.renderer.canvas.ImageLayer', function () {
       target.style.height = '100px';
       document.body.appendChild(target);
       imageExtent = [0, 0, 20, 20];
+      source = new Static({
+        url: `https://openlayers.org/assets/theme/img/logo70.png`,
+        projection: projection,
+        imageExtent: imageExtent,
+      });
+      imageLayer = new ImageLayer({
+        source: source,
+        extent: imageExtent,
+      });
+      sourceCross = new Static({
+        url: `https://openlayers.org/assets/theme/img/logo70.png`,
+        projection: projection,
+        imageExtent: imageExtent,
+        crossOrigin: 'anonymous',
+      });
+      imageLayerCross = new ImageLayer({
+        source: sourceCross,
+        extent: imageExtent,
+      });
       map = new Map({
         pixelRatio: 1,
         target: target,
-        layers: [],
+        layers: [imageLayer, imageLayerCross],
         view: new View({
           projection: projection,
           center: [10, 10],
           zoom: 1,
           maxZoom: 8,
         }),
       });
+      source.on('imageloadend', function () {
```

Comment:
You have to wait until both `source` and `sourceCross` are loaded, e.g.
```js
let loadedCount = 0;
[source, sourceCross].forEach(function (source) {
  source.once('imageloadend', function () {
    loadedCount++;
    if (loadedCount === 2) {
      done();
    }
  });
});
```

Reply:
@ahocevar, really thanks for this hint!

Unfortunately it is not working, still get the test failing where it looks for pixels outside of the layer extent, which should return the empty array if `crossOrigin` is not set https://github.com/boeckMt/openlayers/runs/1403599911?check_suite_focus=true

Here is an Image which shows the [example](https://github.com/boeckMt/openlayers/blob/canvas-layer-renderer-test/examples/hitdetect-image.js) I have created. It is a Map with `100px X 100px` so normally the [pixel values](https://github.com/boeckMt/openlayers/commit/b005f785e2626d72a486ea40c80dbc148ce62185#diff-bc239e9a86b1a23567f4a66ea27677761f98ca67d12dea959bbaa679913abfabR147) **[50, 50] for true** and **[10, 10] for false** which I use to test should work.
![image](https://user-images.githubusercontent.com/3969484/99197735-85e5b900-2794-11eb-8231-1f8ba3bb39b3.png)

I came to the consideration of whether it is a good decision to use the logo as a test image, since this might not always be available and the test then fails. So I tried to only test `getDataAtPixel()` https://github.com/boeckMt/openlayers/blob/test-fix-layer-at-pixel/test/spec/ol/renderer/canvas/imagelayer.test.js#L166
but there also the test fails.

Unfortunately, I have the problem that the tests do not work locally on my machine , because the done function is never called, so I tried to use the GitHub Actions on my repository.

I hope I can fix everything tomorrow and then rebase it to clean commits.